rootchain/sampool/bucket.go

```diff
@@ -0,0 +1,63 @@
+package sampool
+
+import (
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type samSet struct {
+	messages   []rootchain.SAM
+	signatures map[string]bool
+}
+
+func newUniqueSAMs() samSet {
+	return samSet{
+		messages:   make([]rootchain.SAM, 0),
+		signatures: make(map[string]bool),
+	}
+}
+
+func (s *samSet) add(msg rootchain.SAM) {
```

Comment:
I want a comment describing that it's not safe against concurrent access.

Reply:
I think it's pretty apparent from the code